mender-auth/cli/actions.cpp

```diff
@@ -80,6 +82,39 @@ error::Error DoBootstrap(shared_ptr<MenderKeyStore> keystore, const bool force)
 			return err;
 		}
 	}
+	return err;
+}
+
+error::Error DoAuthenticate(context::MenderContext &main_context) {
+	events::EventLoop loop;
+	auto &config = main_context.GetConfig();
+	if (config.server_url == "") {
+		log::Info("No server set in the configuration, skipping authentication");
+		return error::NoError;
+	}
+	http::ClientConfig client_config {
+		config.server_certificate, config.https_client.certificate, config.https_client.key};
+	http::Client client {client_config, loop};
+	auto err = auth_client::FetchJWTToken(
+		client,
+		config.server_url,
+		config.paths.GetKeyFile(),
+		config.paths.GetInventoryScriptsDir(),
+		[&loop, &config](auth_client::APIResponse resp) {
+			log::Info("Got Auth response");
+			if (resp) {
+				log::Info(
+					"Successfully authorized with the server: " + config.server_url + resp.value());
+			} else {
+				log::Error(resp.error().String());
```

Comment:
This will still return `error::NoError`. We should take an `error` by reference in the lambda and set it on actual errors. Or is this by design?

Reply:
As far as I understand we are only supposed to _try_ and authenticate with the server.

Acceptance criteria:
* `Report result on command line.`

Whether it does or not is not so important, as long as we log it.

Reply:
Okey.